gammapy/spectrum/models.py

```diff
@@ -2,24 +2,66 @@
 """Spectral models for Gammapy.
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
+from ..extern.bunch import Bunch
+from . import CountsSpectrum
+import numpy as np
+
 
 __all__ = [
     'SpectralModel',
     'PowerLaw',
     'ExponentialCutoffPowerLaw',
 ]
 
+# Note: Consider to move stuff from _models_old.py here
 
 class SpectralModel(object):
     """Spectral model base class.
     """
+    def __call__(self, energy):
```

Comment:
Change implementation to this:
```
def __call__(self, energy):
    return self.evaluate(energy, **self.parameters)
```

I think your current implementation adds `energy` to `self.parameters` and this is not intended?!
(Try it ... I'm pretty sure from looking at the code.)